mlflow/recipes/steps/train.py

```diff
@@ -325,8 +325,13 @@ def my_warn(*args, **kwargs):
                             f"skipping rebalancing."
                         )
 
-            X_train, y_train = train_df.drop(columns=[self.target_col]), train_df[self.target_col]
+            X_train = train_df.drop(columns=[self.target_col])
 
+            is_array = train_df[self.target_col].apply(lambda x: isinstance(x, np.ndarray))
+            if is_array.all():
+                y_train = np.vstack(train_df[self.target_col].values)
```

Comment:
I think we should validate all values in target_col has the same length array

Reply:
Definitely, thank you